ptf/saidebugcounters.py

```diff
@@ -0,0 +1,2194 @@
+
+    def runTest(self):
+        dc_oid = 0
+
+        self.createRouter()
+        self.createNeighbor()
+        self.createRoute()
+
+        try:
+            drop_list = [SAI_IN_DROP_REASON_UC_DIP_MC_DMAC]
+            dc_oid = self.createDebugCounter(
+                SAI_DEBUG_COUNTER_TYPE_PORT_IN_DROP_REASONS, drop_list)
+            self.assertTrue(dc_oid != 0, "Failed to Create DebugCounter")
+
+            print("\nTest IPv4")
+            self.testPortDebugCounter(
+                self.port0, dc_oid,
+                drop_list,
+                [self.uc_dipv4_mc_dmac_pkt])
+
+            print("Test IPv6")
+            self.testPortDebugCounter(
+                self.port0, dc_oid,
+                drop_list,
+                [self.uc_dipv6_mc_dmac_pkt])
+
+        finally:
+            if dc_oid != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid)
+
+
+@group('debug_counters')
+class PortDropReasonIPHeaderErrorTest(BaseDebugCounterClass):
+    ''' Port Debug Counter for IP header error drop reason test. '''
+
+    def runTest(self):
+        dc_oid = 0
+        try:
+            drop_list = [SAI_IN_DROP_REASON_IP_HEADER_ERROR]
+            dc_oid = self.createDebugCounter(
+                SAI_DEBUG_COUNTER_TYPE_PORT_IN_DROP_REASONS, drop_list)
+            self.assertTrue(dc_oid != 0, "Failed to Create DebugCounter")
+
+            self.testPortDebugCounter(
+                self.port0, dc_oid,
+                drop_list,
+                [self.ihl_pkt])
+
+        finally:
+            if dc_oid != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid)
+
+
+class SwitchDropReasonIPHeaderErrorTest(BaseDebugCounterClass):
+    ''' Switch Debug Counter for IP header error drop reason test. '''
+
+    def runTest(self):
+
+        dc_oid = 0
+        try:
+            drop_list = [SAI_IN_DROP_REASON_IP_HEADER_ERROR]
+            dc_oid = self.createDebugCounter(
+                SAI_DEBUG_COUNTER_TYPE_SWITCH_IN_DROP_REASONS, drop_list)
+            self.assertTrue(dc_oid != 0, "Failed to Create DebugCounter")
+
+            self.testSwitchDebugCounter(
+                dc_oid,
+                drop_list,
+                [self.ihl_pkt])
+
+        finally:
+            if dc_oid != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid)
+
+
+class PortMultiDebugCounters(BaseDebugCounterClass):
+    ''' Port multi Debug Counter test. '''
+
+    def runTest(self):
+        dc_oid1 = 0
+        dc_oid2 = 0
+        dc_oid3 = 0
+        try:
+            drop_list1 = [SAI_IN_DROP_REASON_IP_HEADER_ERROR,
+                          SAI_IN_DROP_REASON_L2_ANY]
+            dc_oid1 = self.createDebugCounter(
+                SAI_DEBUG_COUNTER_TYPE_PORT_IN_DROP_REASONS,
+                drop_list1)
+            self.assertTrue(dc_oid1 != 0, "Failed to Create DebugCounter")
+            self.assertTrue(self.verifyDebugCounterDropList(
+                dc_oid1, drop_list1))
```

Comment:
Inconsistent attribute formatting: may we edit it to have two arguments in one line or each argument for each line?
It should be applied to all cases in this file.

Reply:
no, this is the best we could do as then it is more than > 80 chars